_blinkr/jenkins_blinkr_runner.rb

```diff
@@ -0,0 +1,94 @@
+require_relative '../_docker/lib/process_runner'
+require 'fileutils'
+require 'json'
+
+#
+# This class wraps a number of calls to _docker/control.rb to allow us to run the broken link checks and amalgamate the result into a generalised fail or pass
+#
+class JenkinsBlinkrRunner
+
+  def initialize(host_to_test, control_script_directory, process_runner)
+    @host_to_test = host_to_test
+    @control_script_directory = control_script_directory
+    @process_runner = process_runner
+  end
+
+  #
+  # Run the tests for each of our test profiles
+  #
+  def run_checks
+    execute_checks
+  end
+
+  private
+
+  #
+  # Execute the checks
+  #
+  def execute_checks
+    test_execution_command = build_run_tests_command
+    @process_runner.execute!(test_execution_command)
+    determine_result
+  end
+
+  #
+  # Reads an environment variable, returning its value or nil if it is not set or empty
+  #
+  def read_env_variable(variable_name)
+    return nil if ENV[variable_name].nil? || ENV[variable_name].empty?
+    ENV[variable_name]
+  end
+
+  #
+  # Builds the command to use to execute the tests, including whether or not we should send updates to GitHub and any Cucumber tags
+  # that should be applied
+  #
+  def build_run_tests_command
```

Comment:
Same here about method name (run_blinkr_command rather than run_tests_command) and also the comments in the code.